File: Desktop/Proyectos/Practica modulo 5/appbanco/src/app/persona.ts
export interface Persona {
    id: number;
    nombre?: String;
    apellido?: String;
    usuario?: String;
    password?: String;
    balance?: number;
    salario?: number;
    esGestor?: boolean;
}

export interface Cliente {
    id: number;
    id_gestor: number;
    nombre: String;
    apellido: String;
    usuario: String;
    balance: number;
}

export interface Gestor {
    id: number;
    nombre: String;
    apellido: String;
    usuario: String;
    salario: number;
}

export interface Mensaje {
    id: number;
    mensaje:String;
    us_emisor: String;
    nombre_emisor: String;
    us_receptor: String;
    nombre_receptor: String;
    fecha: String;
}

export interface Deposito {
    id: number;
    id_emisor: number;
    nombre_emisor: String;
    monto: number;
    fecha: String;
}

export interface Transferencia {
    id: number;
    monto: number;
    id_emisor: number;
    nombre_emisor: String
    id_receptor: number;
    nombre_receptor: String
    fecha: String
}

File: Desktop/Aplicacion de banco/banco/src/main/java/com/spring/main/restservice/Persona.java
package com.spring.main.restservice;

public record Persona (int id,int id_gestor ,String nombre, String apellido, String usuario, boolean esGestor, double balance, double salario) {
 
  public Persona (int id, int id_gestor, String nombre, String apellido, String usuario, double balance) {
    this(id, id_gestor, nombre, apellido, usuario, false, balance, 0);
  }
  
  public Persona (int id, String nombre, String apellido, String usuario, double salario) {
    this(id, 0, nombre, apellido, usuario, true, 0, salario);
  }
  
}
